### PR TITLE
ZCS-12731 WCAG Re-opened Issues Part 2

### PR DESCRIPTION
**Pull request by miteshsavani810**
Fixed NVDA announcement for the list item.
- Fixed NVDA announcement when list item is selected by user. Previously, NVDA was not announce as selected when user navigate to checked item.
- Fixed the aria-label with respect to message, conversation, contacts, tasks, briefcase list item such that NVDA announce all relevant information of current focus list item.

See the changes of zm-ajax: https://github.com/Zimbra/zm-ajax/pull/114

### CODE REVIEW

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js [hunk 1]
```diff
@@ -1130,6 +1130,23 @@ function(ev) {
...
+	return headers;
+};
+
+ZmContactSimpleView.prototype._getLabelForField =
+function(item, field) {
+	switch (field) {
+		case ZmItem.F_NAME:
+			return item.sf;
+	}
+	return ZmListView.prototype._getLabelForField.apply(this, arguments);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js:1147]: should this be moved to default case in switch condition?

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 2]
```diff
@@ -198,6 +198,26 @@ function(ex) {
...
+
+ZmTaskListView.prototype._getLabelForField =
+function(item, field) {
+	switch (field) {
+		case ZmItem.F_NAME:
+			return item.name;
+		case ZmItem.F_PRIORITY:
+			return item.isHighPriority ?  ZmMsg.priorityHigh : item.isLowPriority ? ZmMsg.priorityLow : ZmMsg.priorityNormal;
+	}
+	return ZmListView.prototype._getLabelForField.apply(this, arguments);
```
Comment (lavegupta) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js:218]: default case

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 3]
```diff
@@ -362,6 +362,39 @@ function(item) {
...
+ZmDetailListView.prototype._getLabelFieldList =
+function () {
+	var headers = [];
+	headers.push(ZmItem.F_SELECTION);
+	headers.push(ZmItem.F_NAME);
+	headers.push(ZmItem.F_TYPE);
+	headers.push(ZmItem.F_SIZE);
+	headers.push(ZmItem.F_DATE);
+	headers.push(ZmItem.F_SHARES);
+	return headers;
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js:374]: better write it this way?

return [
    ZmItem.F_SELECTION,
    ZmItem.F_NAME,
    ZmItem.F_TYPE,
    ZmItem.F_SIZE,
    ZmItem.F_DATE,
    ZmItem.F_SHARES
];

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 4]
```diff
@@ -198,6 +198,26 @@ function(ex) {
     }
 };
 
+ZmTaskListView.prototype._getLabelFieldList =
+function () {
+	var headers = [];
+	headers.push(ZmItem.F_SELECTION);
+	headers.push(ZmItem.F_NAME);
+	headers.push(ZmItem.F_PRIORITY);
+	return headers;
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js:207]:
```
return [
    ZmItem.F_SELECTION,
    ZmItem.F_NAME,
    ZmItem. F_PRIORITY
];
```

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 5]
```diff
@@ -362,6 +362,39 @@ function(item) {
...
+};
+
+ZmDetailListView.prototype._getLabelForField =
+function(item, field) {
+	switch (field) {
+		case ZmItem.F_NAME:
+			return item.name;
+		case ZmItem.F_TYPE:
+			var itemInfo = ZmMimeTable.getInfo(item.contentType);
+			return itemInfo.desc;
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js:384]: would there be possibility of itemInfo having null or undefined value?

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 6]
```diff
@@ -362,6 +362,39 @@ function(item) {
...
+		case ZmItem.F_DATE:
+			var dateFormatter = AjxDateFormat.getDateTimeInstance(AjxDateFormat.FULL, AjxDateFormat.MEDIUM);
+			var dateStr = dateFormatter.format(item.contentChangeDate);
+			return dateStr;
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js:388]:
```suggestion
			return dateFormatter.format(item.contentChangeDate);
```

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 7]
```diff
@@ -198,6 +198,26 @@ function(ex) {
...
+};
+
+ZmTaskListView.prototype._getLabelForField =
+function(item, field) {
+	switch (field) {
+		case ZmItem.F_NAME:
+			return item.name;
+		case ZmItem.F_PRIORITY:
+			return item.isHighPriority ?  ZmMsg.priorityHigh : item.isLowPriority ? ZmMsg.priorityLow : ZmMsg.priorityNormal;
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js:216]: switch case would be more readable

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 8]
```diff
@@ -362,6 +362,39 @@ function(item) {
...
+		case ZmItem.F_SIZE:
+			return AjxUtil.formatSize(item.size);
+		case ZmItem.F_SHARES:
+			return item.modifier;
+	}
+
+	return ZmListView.prototype._getLabelForField.apply(this, arguments);
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Added.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Added.

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js [hunk 9]
```diff
@@ -1130,6 +1130,23 @@ function(ev) {
 	}
 };
 
+ZmContactSimpleView.prototype._getLabelFieldList =
+function () {
+	var headers = [];
+	headers.push(ZmItem.F_SELECTION);
+	headers.push(ZmItem.F_NAME);
+	return headers;
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js]: Done.

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Fixed.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 4 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Fixed

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js (hunk 5 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Addressed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js (hunk 6 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Fixed.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 7 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Addressed.

**Review by silentsakky — APPROVED** [review]

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 10]
```diff
@@ -198,6 +198,33 @@ function(ex) {
...
+		case ZmItem.F_NAME:
+			return item.name;
+		case ZmItem.F_PRIORITY:
+			if (item.isHighPriority) {
+				return ZmMsg.priorityHigh;
+			} else if (item.isLowPriority) {
+				return ZmMsg.priorityLow;
+			} else {
+				return ZmMsg.priorityNormal;
+			}
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js:222]:
```suggestion
			if (item.isHighPriority) {
				return ZmMsg.priorityHigh;
			}
			if (item.isLowPriority) {
				return ZmMsg.priorityLow;
			}
			return ZmMsg.priorityNormal;
```

Comment (silentsakky) [reply WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js:222]: it's either or condition so current version looks better

**Review by HimanshuSojitraSynacor — APPROVED** [review]

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 10 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Done.

**Review by lavegupta — APPROVED** [review]

**Pull request merged**